File: app/src/main/java/com/example/usha/profile/ProfileViewModel.java
package com.example.usha.profile;

import android.os.Bundle;
import android.view.View;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ProfileViewModel extends ViewModel {
    private final MutableLiveData<String> user = new MutableLiveData<String>("ProfileFragment");


}

File: app/src/main/java/com/headingWarm/usha/profile/ProfileViewModel.kt
package com.headingWarm.usha.profile

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class ProfileViewModel(val model: ProfileModel): ViewModel() {

    val name: String get() = model.name
    val email: String get() = model.email


    fun clickSettingBtn(){
        model.goToSettingFrag()
    }


    class ProfileVMFac(val model: ProfileModel): ViewModelProvider.Factory{
        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            return ProfileViewModel(model) as T
        }
    }
}